### PR TITLE
Docstring fixes for galaxy.exceptions and galaxy.jobs.metrics.

### PR DESCRIPTION
**Pull request by jmchilton**
xref https://github.com/galaxyproject/galaxy-lib/pull/21.

### CODE REVIEW

File: lib/galaxy/exceptions/__init__.py [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
 """
 
 from ..exceptions import error_codes
 
 
 class MessageException( Exception ):
-    """
-    Exception to make throwing errors from deep in controllers easier.
-    """
+    """Most generic Galaxy exception - indicates merely that some exceptional condition happend."""
```
Comment (nsoranzo) [inline lib/galaxy/exceptions/__init__.py]: s/happend/happened/

**Comment by nsoranzo** [comment]
It would also be nice to have the definition of `[testenv:py27-lint-docstring]` in `tox.ini` (not in Travis).

File: lib/galaxy/jobs/metrics/collectl/cli.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+"""This module describes :class:`CollectlCli` - an asbtraction for building collectl command lines."""
```
Comment (nsoranzo) [inline lib/galaxy/jobs/metrics/collectl/cli.py]: s/asbtraction/abstraction/

File: lib/galaxy/jobs/metrics/__init__.py [hunk 3]
```diff
@@ -1,3 +1,15 @@
+"""This module defines the job metrics collection framework for Galxay jobs.
```
Comment (nsoranzo) [inline lib/galaxy/jobs/metrics/__init__.py]: s/Galxay/Galaxy/

File: lib/galaxy/jobs/metrics/__init__.py [hunk 4]
```diff
@@ -1,3 +1,15 @@
...
+
+The framework consists of two parts - the :class:`JobMetrics` class and
+individual :class:`JobInstrumenter` plugins.
+
+A :class:`JobMetrics` object reads any number of plugins from a configuration
+source such as an XML file, a YAML file, or dictionary.
```
Comment (nsoranzo) [inline lib/galaxy/jobs/metrics/__init__.py]: s/or dictionary/or a dictionary/

File: lib/galaxy/jobs/metrics/__init__.py [hunk 5]
```diff
@@ -1,3 +1,15 @@
...
+
+The :class:`JobInstrumenter` plugin objects each describes how to inject a bits
```
Comment (nsoranzo) [inline lib/galaxy/jobs/metrics/__init__.py]: `The :class:`JobInstrumenter` plugin objects each describes`: sounds a bit convoluted, what about `Each :class:`JobInstrumenter` plugin object describes` ?

**Review by nsoranzo — CHANGES_REQUESTED** [review]

**Comment by jmchilton** [comment]
@nsoranzo - Adding a py27-lint-docstring feels like it introduces more politics, how about you merge this and I promise to follow up with a PR on that point?

File: lib/galaxy/jobs/metrics/collectl/cli.py [hunk 6]
```diff
@@ -1,3 +1,4 @@
+"""This module describes :class:`CollectlCli` - an abstracton for building collectl command lines."""
```
Comment (nsoranzo) [inline lib/galaxy/jobs/metrics/collectl/cli.py]: s/abstracton/abstraction/

**Comment by nsoranzo** [comment]
> Adding a py27-lint-docstring feels like it introduces more politics, how about you merge this and I promise to follow up with a PR on that point?

No problem, but there are still 2 small unresolved issues and a new one you just introduced :laughing:

**Comment by jmchilton** [comment]
:crying_cat_face:

**Comment by martenson** [comment]
but hey, you are still moving forward :)

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**